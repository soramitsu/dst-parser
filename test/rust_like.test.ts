import { parseFile } from './test_util'
import { describe, expect, it } from 'vitest'

describe('Rust / Java / C syntax tests', () => {
    it('handles a code section', () => {
        let parsed = parseFile('code_section.rs')
        expect(parsed).toMatchSnapshot()
    })
  
    it('handles the indentation levels', () => {
        let parsed = parseFile('code_section_2.rs')
        expect(parsed).toMatchSnapshot()
    })

    it('handles a nested fragment escape normally', () => {
        let parsed = parseFile('nested_fragment_escape.c')
        expect(parsed).toMatchSnapshot()
    })

    it('handles a nested escape with a fragment placed inside the escape text', () => {
        let parsed = parseFile('nested_escape_test.c')
        expect(parsed).toMatchSnapshot()
    })
})