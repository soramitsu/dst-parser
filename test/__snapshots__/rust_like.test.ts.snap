// Vitest Snapshot v1

exports[`Rust / Java / C syntax tests > handles a code section 1`] = `
{
  "RustStyleFragment": "println!(\\"A normal Rust-style fragment\\");",
}
`;

exports[`Rust / Java / C syntax tests > handles a nested escape with a fragment placed inside the escape text 1`] = `
{
  "TestA": "This part is displayed",
}
`;

exports[`Rust / Java / C syntax tests > handles a nested fragment escape normally 1`] = `
{
  "TestA": "This part
is displayed",
  "TestB": "is displayed",
}
`;

exports[`Rust / Java / C syntax tests > handles the indentation levels 1`] = `
{
  "RustStyleFragment": "fn main() {
    println!(\\"A normal Rust-style fragment\\");
}",
}
`;
